Shader compiler errors on NVIDIA hardware

I think we're trying to use **unpackHalf2x16** on an OpenGL 4.1 platform without enabling the appropriate shading language extension.  Details follow.

When examining a glTF with additional lights (thus requiring froxelization) on a machine with GeForce RTX 2060 + Windows 10, I see the following:

> FEngine (64 bits) created at 00000128E6FE1080 (threading is enabled)
> FEngine resolved backend: OpenGL
> NVIDIA Corporation, GeForce RTX 2060/PCIe/SSE2, 4.1.0 NVIDIA 461.09, 4.10 NVIDIA via Cg compiler
> OS version: 0
> Froxel: 1024x640 / 40x40
> Froxel: 26x16x16 = 6656 (1536 lost)
> Froxel: 614x640 / 32x32
> Froxel: 20x20x16 = 6400 (1792 lost)
> Compilation error in fragment shader "Material":
> "0(4) : warning C7508: extension GOOGLE_cpp_style_line_directive not supported
> 0(3139) : error C7532: global function unpackHalf2x16 requires "#version 420" or later
> 0(3139) : error C0000: ... or #extension GL_ARB_shading_language_packing : enable

--- a/filament/backend/src/opengl/OpenGLProgram.cpp
+++ b/filament/backend/src/opengl/OpenGLProgram.cpp
@@ -94,7 +94,7 @@ vec2 unpackHalf2x16(highp uint v) {
 }
 
 )"};
-                // a good point for insertion is just before the first occurence of an uniform block
+                // a good point for insertion is just before the first occurrence of an uniform block
                 auto pos = temp.find("layout(std140)");
                 if (pos != std::string_view::npos) {
                     temp.insert(pos, unpackHalf2x16);

--- a/libs/filamat/src/GLSLPostProcessor.cpp
+++ b/libs/filamat/src/GLSLPostProcessor.cpp
@@ -344,6 +344,11 @@ void GLSLPostProcessor::fullOptimization(const TShader& tShader,
         CompilerGLSL glslCompiler(move(spirv));
         glslCompiler.set_common_options(glslOptions);
 
+        if (tShader.getStage() == EShLangFragment && !glslOptions.es) {
+            // enable GL_ARB_shading_language_packing if available
+            glslCompiler.add_header_line("#extension GL_ARB_shading_language_packing : enable");
+        }
+
         if (tShader.getStage() == EShLangFragment && glslOptions.es) {
             for (auto i : config.glsl.subpassInputToColorLocation) {
                 glslCompiler.remap_ext_framebuffer_fetch(i.first, i.second, true);

--- a/libs/filamat/src/shaders/CodeGenerator.cpp
+++ b/libs/filamat/src/shaders/CodeGenerator.cpp
@@ -64,6 +64,7 @@ io::sstream& CodeGenerator::generateProlog(io::sstream& out, ShaderType type,
                 out << "#version 450 core\n\n";
             } else {
                 out << "#version 410 core\n\n";
+                out << "#extension GL_ARB_shading_language_packing : enable\n\n";
             }
             break;
     }

